support-lib/jni/Marshal.hpp

```diff
@@ -178,6 +178,25 @@ namespace djinni
         }
     };
 
+    struct WString
+    {
+        using CppType = std::wstring;
+        using JniType = jstring;
+
+        using Boxed = WString;
+
+        static CppType toCpp(JNIEnv* jniEnv, JniType j)
+        {
+            assert(j != nullptr);
+            return jniWStringFromString(jniEnv, j);
+        }
+
+        static LocalRef<JniType> fromCpp(JNIEnv* jniEnv, const CppType& c)
+        {
+            return {jniEnv, jniStringFromWString(jniEnv, c.c_str())};
```

Comment:
The `c_str()` here shouldn't be necessary (and it will break handling of strings containing U+0000)